### ISSUE TITLE
Miscompiled code

### ISSUE THREAD

**Issue by nurmukhametov** [ispc/ispc#2552]
```c
uniform int boo() {
  const uniform int a = 2;
  uniform const int * uniform const p = &a;
  uniform int x = a + *p;
  return x;
}

void foo() {
    print("%", boo());
}
```

According to https://godbolt.org/z/hP5aEfxjd is miscompiled as below:
```c
__fast_masked_vload:                    # @__fast_masked_vload
        xor     eax, eax
        ret
boo___:                                 # @boo___
foo___:                                 # @foo___
```

Such example:
```c
#include "./test_static.isph"
uniform int foo() {
  const uniform int a = 2;
  uniform const int * uniform const p = &a;
  uniform int x = a + *p;
  return x;
}

task void f_v(uniform float RET[]) {
    RET[programIndex] = (float)foo();
}

task void result(uniform float RET[]) {
    RET[programIndex] = 4.000000;
}
``` 
fails
```bash
/***/core$ time -p ./run_tests.py --target=avx2-i32x8 --arch=x86-64 ./2291_4.ispc -v
Testing ISPC compiler: ../ispc-releases/ispc-v1.20.0-linux/bin/ispc
Testing ISPC target: avx2-i32x8
Testing ISPC arch: x86-64

Using test compiler: Intel(r) Implicit SPMD Program Compiler (Intel(r) ISPC), 1.20.0 (build commit 14bd04aa7e68cd33 @ 20230506, LLVM 15.0.7)
Using C/C++ compiler: clang version 15.0.7 (***/llvm-build/llvm-project 8dfdcc7b7bf66834a761bd8de445840ef68e4d1a)

Running 1 jobs in parallel. Running 1 tests.
Running: /***/ispc-releases/ispc-v1.20.0-linux/bin/ispc  --pic --woff /***/core/2291_4.ispc -o ./2291_4.ispc.o --arch=x86-64 --target=avx2-i32x8 -DTEST_SIG=0 -O2 -h /***/core/2291_4.ispc.h
Running: clang++ -O2 -I. -m64 test_static.cpp -DTEST_SIG=0 -DTEST_WIDTH=8 ./2291_4.ispc.o -o ./2291_4.ispc.run -fPIE -DTEST_HEADER="</l***/core/2291_4.ispc.h>"
Running:  ./2291_4.ispc.run
Test ./2291_4.ispc failed (return code -9)
 Done 1 / 1 [./2291_4.ispc]

Executed 1 / 1 (0 skipped)

PASSRATE (0/1) = 0%

0 / 1 tests PASSED
0 / 1 tests FAILED compilation
1 / 1 tests FAILED execution
        ./2291_4.ispc
0 / 1 tests SKIPPED

real 35.58
user 2.44
sys 8.28
``` 

This code is inspired by ICE in https://github.com/ispc/ispc/issues/2291

**Comment by zephyr111** [comment]
With the PR #3424, the programs correctly print 4 on Linux, so it fixes this issue. 

So far, the generated LLVM IR code was the following:

```
define noundef i32 @boo___(<8 x i32> %__mask) local_unnamed_addr #0 {
allocas:
  unreachable
}

; Function Attrs: noreturn nounwind optsize
define void @foo___(<8 x i32> %__mask) local_unnamed_addr #1 {
allocas:
  unreachable
}
```

Please note the `unreachable` code (so the assembly code was empty).

While with the new PR, it is:

```
define noundef i32 @boo___(<8 x i32> %__mask) local_unnamed_addr #0 {
allocas:
  ret i32 4
}

; Function Attrs: nounwind optsize
define void @foo___(<8 x i32> %__mask) local_unnamed_addr #1 {
allocas:
  %print_arg_ptrs = alloca [1 x ptr], align 32
  %print_arg = alloca i32, align 4
  store i32 4, ptr %print_arg, align 4
  store ptr %print_arg, ptr %print_arg_ptrs, align 32
  %0 = icmp slt <8 x i32> %__mask, zeroinitializer
  %1 = bitcast <8 x i1> %0 to i8
  %v64.i = zext i8 %1 to i64
  call fastcc void @__do_print(i64 %v64.i, ptr nonnull %print_arg_ptrs) #7
  ret void
}
```

There are a lot of IR code after `@foo___`, apparently because of the `print` (e.g. functions like `@__do_print`, `@_ZN9ArgWriterC2EPKPKvim`, etc.). Such part of the IR code is not present without `foo`.
